Automattic/woocommerce-payments
# SCA support for subscription renewals

Fixes #2002 
Fixes #2113 
Fixes #2114 

This PR is big, but do not despair. Most of it is just adding the email templates.

#### Changes proposed in this Pull Request

- Add an admin-facing email that's triggered when a Subscription renewal fails due to SCA. That email will replace the built-in email that's sent by WC Subscriptions.
- Add a customer-facing email that's triggered when a Subscription renewal fails due to SCA. That email has a link that the customer can click to immediately authenticate the payment.
- Tweak the Pay For Order page so the SCA modal automatically appears.

#### To test

- Install and activate WC Subscriptions.
- Create a Subscription product. It doesn't matter if it has a free trial.
- As a customer, buy the subscription product with an SCA test card. I recommend `4000002760003184` since that one will ask for SCA auth on every renewal payment.
- Go to `WP-Admin -> WooCommerce -> Subscriptions`, click on the Subscription you just created, and on the `Subscription Actions` metabox select `Process renewal`.
- Note that the Subscription is set to `On Hold` status, the renewal order is set to `Failed` status, and that you received 1 email as an admin and 1 email as a customer (I recommend installing the `WP Mail Logging` plugin).
- Click the `Authorize the payment` link on the email you received as a customer. You should see a bare-bones Pay For Order page and the SCA modal should pop up automatically.
- If you accept the modal, you'll get to the `Thank you` page, and the Subscription will be on `Active` status again.
- If you reject authentication instead, you'll see an error message. If you refresh the page, you'll see the default `Pay For Order` page, and you'll be able to pay for the renewal manually.
